src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -794,16 +797,41 @@ public RubyNode visitCaseNode(CaseParseNode node) {
 
                 // Create the if node
                 final RubyNode thenNode = translateNodeOrNil(sourceSection, when.getBodyNode());
-                final IfElseNode ifNode = new IfElseNode(conditionNode, thenNode, elseNode);
 
-                // This if becomes the else for the next if
-                elseNode = ifNode;
+                conditions[n] = conditionNode;
+                bodies[n] = thenNode;
             }
 
-            final RubyNode ifNode = elseNode;
+            final WhenParseNode firstWhen = (WhenParseNode) node.getCases().get(0);
+            final boolean anyCase = firstWhen instanceof WhenOneArgParseNode
+                    && firstWhen.getExpressionNodes() instanceof SymbolParseNode
+                    && ((SymbolParseNode) firstWhen.getExpressionNodes()).getName().equals("truffleruby_any_case");
 
-            // A top-level block assigns the temp then runs the if
-            ret = sequence(sourceSection, Arrays.asList(assignTemp, ifNode));
+            if (anyCase) {
+                final BooleanCastNode[] conditionCasts = new BooleanCastNode[conditions.length];
+                for (int n = 0; n < conditionCasts.length; n++) {
```

Comment:
We'd likely want to ignore the `firstWhen` here.

Reply:
With the marker? Yes but I want a better way to do that.